packages/appstore/src/types/common.types.ts

```diff
@@ -189,24 +189,41 @@ export type TLinkedTo = {
 
 export type TWalletAccount = NonNullable<ReturnType<typeof useWalletsList>['data']>[number];
 export type TWalletInfo = NonNullable<ReturnType<typeof useAvailableWallets>['data']>[number];
-export type TWalletButton = {
-    name: Parameters<ReturnType<typeof useStore>['traders_hub']['setWalletModalActiveTab']>[0];
-    text: string;
-    icon: string;
-    action: () => void;
-};
 
 export type TTransferAccount = {
-    active_wallet_icon?: string;
+    active_wallet_icon: string | undefined;
     account_type?: 'wallet' | 'trading' | 'dxtrade' | 'mt5' | 'derivez' | 'binary';
-    balance?: number;
+    balance: number;
     currency?: string;
-    display_currency_code?: string;
+    display_currency_code: string | undefined;
     gradient_class?: `wallet-card__${string}`;
     icon?: string | undefined;
-    is_demo?: boolean;
+    is_demo: boolean;
     loginid?: string;
     mt5_market_type?: 'all' | 'financial' | 'synthetic';
-    shortcode?: string;
-    type?: 'fiat' | 'crypto' | 'demo';
+    shortcode: string | undefined;
+    type: 'fiat' | 'crypto' | 'demo';
+};
+
+export type TMessageItem =
+    | {
+          variant: 'base';
+          key: string;
+          type: 'info' | 'error' | 'success';
+          message: string | JSX.Element;
+      }
+    | {
+          variant: 'with-action-button';
+          onClickHandler: VoidFunction;
+          button_label: string;
+          key: string;
+          type: 'info' | 'error' | 'success';
+          message: string | JSX.Element;
+      };
```

Comment:
```suggestion
export type TMessageItem =
    | {
          variant: 'base';
          key: string;
          type: 'info' | 'error' | 'success';
          message: string | JSX.Element;
      }
    | {
          variant: 'with-action-button';
          onClickHandler: VoidFunction;
          button_label: string;
          key: string;
          type: 'info' | 'error' | 'success';
          message: string | JSX.Element;
      };
```
Please think about the structure of the MessageItem type. Doing it this way seems like you are just adding more types to solve your typescript error. Think about one structure that your data structure should have to make your component work and if you have other properties like `OnClick` or `label` there is always a generic type u can extend from

Reply:
hey @yashim-deriv 
i'm using this approach (Discriminated Unions) with current structure in order not to make some of the props optional and make all of them required based on the variant of the component that you want to render (with or without button). 
when the developer wants to use MessageItem with button, we remind him that you should not forget to pass `onClick` and `Label` props when you want component with button.